cfme/tests/test_rest.py

```diff
@@ -69,19 +72,144 @@ def provision_data(
     return result
 
 
+@pytest.fixture(scope='function')
+def service_catalogs(request, rest_api):
+    name = fauxfactory.gen_alphanumeric()
+    rest_api.collections.service_catalogs.action.add(
+        name=name,
+        description="description_{}".format(name),
+        service_templates=[]
+    )[0]
+    wait_for(
+        lambda: rest_api.collections.service_catalogs.find_by(name=name),
+        num_sec=180,
+        delay=10,
+    )
+
+    scls_data = [{
+        "name": "name_{}_{}".format(name, index),
+        "description": "description_{}_{}".format(name, index),
+        "service_templates": []
+    } for index in range(1, 5)]
+    scls = rest_api.collections.service_catalogs.action.add(*scls_data)
+
+    @request.addfinalizer
+    def _finished():
+        scls = [_ for _ in rest_api.collections.service_catalogs]
+        if len(scls) != 0:
+            rest_api.collections.service_catalogs.action.delete(*scls)
+
+    return scls
+
+
+@pytest.mark.usefixtures("logged_in")
+@pytest.fixture(scope='function')
+def dialog():
+    dialog = "dialog_{}".format(fauxfactory.gen_alphanumeric())
+    element_data = dict(
+        ele_label="ele_{}".format(fauxfactory.gen_alphanumeric()),
+        ele_name=fauxfactory.gen_alphanumeric(),
+        ele_desc="my ele desc",
+        choose_type="Text Box",
+        default_text_box="default value"
+    )
+    service_dialog = ServiceDialog(
+        label=dialog,
+        description="my dialog",
+        submit=True,
+        cancel=True,
+        tab_label="tab_{}".format(fauxfactory.gen_alphanumeric()),
+        tab_desc="my tab desc",
+        box_label="box_{}".format(fauxfactory.gen_alphanumeric()),
+        box_desc="my box desc")
+    service_dialog.create(element_data)
+    return service_dialog
+
+
+@pytest.mark.usefixtures("logged_in")
+@pytest.fixture(scope='function')
+def service_templates(request, rest_api, dialog):
+    try:
+        if current_version() <= "5.5":
```

Comment:
Also, is the `try` block needed for the whole section or is it there only because of `s_tpls[0]` on line 148? If so, we definitely want to move the `try` statement further down to avoid enclosing unnecessary lines in it.